File: benchmarks/perf/calls.c
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

static inline int64_t checkedAdd(int64_t left, int64_t right) {
    int64_t result;
    if (__builtin_add_overflow(left, right, &result)) {
        abort();
    }
    return result;
}

static int64_t adjust(int64_t value) {
    return checkedAdd(value, 3);
}

int main(void) {
    int64_t total = 0;
    for (int64_t value = 0; value < 100000000; ++value) {
        total = checkedAdd(total, adjust(value));
    }
    printf("%" PRId64 "\n", total);
    return 0;
}

File: benchmarks/perf/calls.rs
fn adjust(value: i64) -> i64 {
    value.checked_add(3).expect("benchmark overflow")
}

fn main() {
    let mut total: i64 = 0;
    for value in 0_i64..100_000_000 {
        total = total.checked_add(adjust(value)).expect("benchmark overflow");
    }
    println!("{total}");
}
